tests/unit/utils/sparqlhandler/test_data_cube_inspector.py

```diff
@@ -908,6 +908,177 @@ def test_get_columns_for_component_attribute_pivoted():
+    assert result["qb-id-10003.csv#measure/some-other-measure"] == "Some Other Measure"
+    assert result["qb-id-10003.csv#measure/some-measure"] == "Some Measure"
+
+
+def test_get_measure_uris_and_labels_standard_multi_measure():
+    """
+    This test checks that the measure URIs and labels can successfully be
+    retrieved from a standard shape multi measure data set, producing the expected
+    results and without errors.
+    """
+    path_to_json_file = (
+        _test_case_base_dir
+        / "multi-unit_multi-measure"
+        / "new"
+        / "alcohol-bulletin.csv-metadata.json"
+    )
+    csvw_rdf_manager = get_csvw_rdf_manager(path_to_json_file)
+    data_cube_inspector = get_data_cube_inspector(path_to_json_file)
+    primary_catalog_metadata = (
+        csvw_rdf_manager.csvw_inspector.get_primary_catalog_metadata()
+    )
+    csv_url = data_cube_inspector.get_cube_identifiers_for_data_set(
+        primary_catalog_metadata.dataset_uri
+    ).csv_url
+
+    result = data_cube_inspector.get_measure_uris_and_labels(csv_url)
+
+    assert len(result) == 9
+    assert (
+        result["alcohol-bulletin.csv#measure/alcohol-duty-receipts"]
+        == "alcohol-duty-receipts"
+    )
+    assert (
+        result["alcohol-bulletin.csv#measure/beer-duty-receipts"]
+        == "beer-duty-receipts"
+    )
+    assert (
+        result["alcohol-bulletin.csv#measure/cider-duty-receipts"]
+        == "cider-duty-receipts"
+    )
+    assert result["alcohol-bulletin.csv#measure/clearances"] == "clearances"
+    assert (
+        result["alcohol-bulletin.csv#measure/clearances-of-alcohol"]
+        == "clearances-of-alcohol"
+    )
+    assert (
+        result["alcohol-bulletin.csv#measure/production-volume"] == "production-volume"
+    )
+    assert (
+        result["alcohol-bulletin.csv#measure/production-volume-alcohol"]
+        == "production-volume-alcohol"
+    )
+    assert (
+        result["alcohol-bulletin.csv#measure/spirits-duty-receipts"]
+        == "spirits-duty-receipts"
+    )
+    assert (
+        result["alcohol-bulletin.csv#measure/wine-duty-receipts"]
+        == "wine-duty-receipts"
+    )
+
+
+def test_get_attribute_value_uris_and_labels():
+    """
+    This test checks that retrieving the attribute value URIs and labels from a data cube
+    works successfully and produces the expected results.
+    """
+    path_to_json_file = (
+        _test_case_base_dir
+        / "multi-attribute-resource-values"
+        / "multi-attribute.csv-metadata.json"
+    )
+    csvw_rdf_manager = get_csvw_rdf_manager(path_to_json_file)
+    data_cube_inspector = get_data_cube_inspector(path_to_json_file)
+    primary_catalog_metadata = (
+        csvw_rdf_manager.csvw_inspector.get_primary_catalog_metadata()
+    )
+    csv_url = data_cube_inspector.get_cube_identifiers_for_data_set(
+        primary_catalog_metadata.dataset_uri
+    ).csv_url
+
+    result = data_cube_inspector.get_attribute_value_uris_and_labels(csv_url)
```

Comment:
What happens when we call this function but the CSV has an ExistingQbAttributeResource column?